plugins/inventory/xen_orchestra.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from jsonrpc_websocket import Server
+from ansible.errors import AnsibleError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+import asyncio
+import json
+import os
+import re
+
+
+DOCUMENTATION = '''
+    name: xen_orchestra
+    short_description: Xen Orchestra inventory source
+    author:
+        - Dom Del Nano (@ddelnano) <ddelnano@gmail.com>
+        - Samori Gorse (@shinuza) <samorigorse@gmail.com>
+    requirements:
+        - jsonrpc_websocket >= 3.0
+    description:
+        - Get inventory hosts from a Xen Orchestra deployment.
+        - 'Uses a configuration file as an inventory source, it must end in C(.xen_orchestra.yml) or C(.xen_orchestra.yaml)'
+    extends_documentation_fragment:
+        - constructed
+        - inventory_cache
+    options:
+        plugin:
+            description: The name of this plugin, it should always be set to C(community.general.xen_orchestra) for this plugin to recognize it as it's own.
+            required: yes
+            choices: ['xen_orchestra', 'community.general.xen_orchestra']
+            type: str
+        api_host:
+            description:
+                - Api host to XOA api.
+                - If the value is not specified in the inventory configuration, the value of environment variable C(XO_API_HOST) will be used instead.
+            default: '192.168.1.123'
+            type: str
+            env:
+                - name: XO_API_HOST
+        user:
+            description:
+                - Xen Orchestra user.
+                - If the value is not specified in the inventory configuration, the value of environment variable C(XO_USER) will be used instead.
+            required: yes
+            type: str
+            env:
+                - name: XO_USER
+        password:
+            description:
+                - Xen Orchestra password.
+                - If the value is not specified in the inventory configuration, the value of environment variable C(XO_PASSWORD) will be used instead.
+            required: yes
+            type: str
+            env:
+                - name: XO_PASSWORD
+        validate_certs:
+            description: Verify SSL certificate if using HTTPS.
+            type: boolean
+            default: yes
+        use_ssl:
+            description: Use wss when connecting to the Xen Orchestra API
+            type: boolean
+            default: yes
```

Comment:
```suggestion
            default: true
```
Always use `true` and `false` as YAML booleans (appears more than once).